build/pluginctl/logs_test.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+	"time"
+)
+
+func TestCheckOldestEntry(t *testing.T) {
+	for name, tc := range map[string]struct {
+		logs           []string
+		oldest         string
+		expectedLogs   []string
+		expectedOldest string
+		expectedAllNew bool
+	}{
+		"nil logs": {
+			logs:           nil,
+			oldest:         "oldest",
+			expectedLogs:   nil,
+			expectedOldest: "oldest",
+			expectedAllNew: false,
+		},
+		"empty logs": {
+			logs:           []string{},
+			oldest:         "oldest",
+			expectedLogs:   nil,
+			expectedOldest: "oldest",
+			expectedAllNew: false,
+		},
+		"no new entries, one old entry": {
+			logs:           []string{"old"},
+			oldest:         "old",
+			expectedLogs:   []string{},
+			expectedOldest: "old",
+			expectedAllNew: false,
+		},
+		"no new entries, multipile old entries": {
+			logs:           []string{"old1", "old2", "old3"},
+			oldest:         "old3",
+			expectedLogs:   []string{},
+			expectedOldest: "old3",
+			expectedAllNew: false,
+		},
+		"one new entry, no old entry": {
+			logs:           []string{"new"},
+			oldest:         "old",
+			expectedLogs:   []string{"new"},
+			expectedOldest: "new",
+			expectedAllNew: true,
+		},
+		"multipile new entries, no old entry": {
+			logs:           []string{"new1", "new2", "new3"},
+			oldest:         "old",
+			expectedLogs:   []string{"new1", "new2", "new3"},
+			expectedOldest: "new3",
+			expectedAllNew: true,
+		},
+		"one new entry, one old entry": {
+			logs:           []string{"old", "new"},
+			oldest:         "old",
+			expectedLogs:   []string{"new"},
+			expectedOldest: "new",
+			expectedAllNew: false,
+		},
+		"one new entry, multipile old entries": {
+			logs:           []string{"old1", "old2", "old3", "new"},
+			oldest:         "old3",
+			expectedLogs:   []string{"new"},
+			expectedOldest: "new",
+			expectedAllNew: false,
+		},
+		"multipile new entries, ultipile old entries": {
+			logs:           []string{"old1", "old2", "old3", "new1", "new2", "new3"},
+			oldest:         "old3",
+			expectedLogs:   []string{"new1", "new2", "new3"},
+			expectedOldest: "new3",
+			expectedAllNew: false,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			logs, oldest, allNew := checkOldestEntry(tc.logs, tc.oldest)
+
+			if allNew != tc.expectedAllNew {
+				t.Logf("expected allNew: %v, got %v", tc.expectedAllNew, allNew)
+				t.Fail()
+			}
+			if oldest != tc.expectedOldest {
+				t.Logf("expected oldest: %v, got %v", tc.expectedOldest, oldest)
+				t.Fail()
+			}
+
+			compareSlice(t, tc.expectedLogs, logs)
+		})
+	}
+}
+
+func TestFilterLogEntries(t *testing.T) {
```

Comment:
Does this contain the fix at https://github.com/mattermost/mattermost-plugin-starter-template/pull/197?